crates/nargo/src/cli/test_cmd.rs

```diff
@@ -0,0 +1,89 @@
+use std::io::Write;
+
+use acvm::ProofSystemCompiler;
+use clap::ArgMatches;
+use noirc_driver::{CompiledProgram, Driver};
+use noirc_frontend::node_interner::FuncId;
+
+use crate::{errors::CliError, resolver::Resolver};
+
+use super::{add_std_lib, prove_cmd::parse_and_solve_witness};
+
+pub(crate) fn run(args: ArgMatches) -> Result<(), CliError> {
+    let args = args.subcommand_matches("test").unwrap();
+    let test_name = args.value_of("test_name").unwrap_or("");
+    let allow_warnings = args.is_present("allow-warnings");
+    run_tests(test_name, allow_warnings)
+}
+
+fn run_tests(test_name: &str, allow_warnings: bool) -> Result<(), CliError> {
+    let backend = crate::backends::ConcreteBackend;
+
+    let package_dir = std::env::current_dir().unwrap();
+    let mut driver = Resolver::resolve_root_config(&package_dir, backend.np_language())?;
+    add_std_lib(&mut driver);
+
+    if driver.check_crate(allow_warnings).is_err() {
+        std::process::exit(1);
```

Comment:
Not blocking: Can we open up an issue about avoiding the usage of `std::process:exit(1)`? Mainly because std::process::exit prevents drops from happening.

Reply:
Opened #729